### ISSUE TITLE
Particle Crash

### ISSUE THREAD

**Issue by WesCook** [whichonespink44/Appalachia#160]
Using Appalachia-1.10.2-0.1.0-UNSTABLE-ALPHA-3.jar, I get an NPE "Exception while adding particle".

I've seen this numerous times while exploring RTG worlds.

Crash Log: http://pastebin.com/Vji1mwL4

**Comment by whichonespink44** [comment]
Looks like we need to handle the possibility of `worldIn` and/or `pos` being null in AppalachiaBlockLeaves.ColourHandler#colorMultiplier.

Potential solutions:

https://github.com/Elix-x/Colourful-Blocks/commit/343504e8d43857e19f2b87571038a2a9fd26a390

https://github.com/asiekierka/Computronics/commit/8f56d63fc53cdbc08cce29903394f1318386ed47

**Comment by srs-bsns** [comment]
The method clearly states that null values can be passed :P
```java
public int colorMultiplier(IBlockState state, @Nullable IBlockAccess worldIn, @Nullable BlockPos pos, int tintIndex)
```
So you'll have to check that they're not null before you pass them in the call to [`BiomeColorHelper#getFoliageColorAtPos`](https://github.com/Team-RTG/Appalachia/blob/1.10.2-master/src/main/java/appalachia/block/leaves/AppalachiaBlockLeaves.java#L141) which has no null checking.

Edit: Also, `AppalachiaBlockLeaves$ColourHandler` should probably be annotated with `@SideOnly(Side.CLIENT)` since it's client-side only.

**Comment by whichonespink44** [comment]
Fixed in 40142c8 and will be in the next release.
